torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4672,21 +4698,6 @@ def method_tests():
         ('float_power', torch.rand(S, S, S) + 1e-3, (uniform_scalar(0.1),), 'scalar_broadcast_rhs'),
         ('float_power', uniform_scalar(1e-3, requires_grad=True), (torch.rand(S, S, S) + 0.1,), 'scalar_broadcast_lhs'),
         ('float_power', torch.rand(S, S, S) + 1e-3, (3.14,), 'constant'),
-        ('transpose', (1, 2, 3), (1, 2), 'dim', (False,), [0, 1]),
```

Comment:
[0, 1] at the end corresponds to the following line
https://github.com/pytorch/pytorch/blob/f5a7b2e641a124751dfa3fdf1926eb4135d3aad5/test/test_autograd.py#L5390-L5393

For entry `('transpose', (1, 2, 3), (1, 2), 'dim', (False,), [0, 1])` creates tests with args being  `[(-1, -2), (-1, 2), (1, -2), (1, 2)]`

We directly add these to cases 
https://github.com/pytorch/pytorch/blob/d4875aa5dd6499b31e2a0d01466d0a2382004d6b/torch/testing/_internal/common_methods_invocations.py#L711-L718

Had added it with similar to machinery to test_autograd test generator but adding directly seemed more readable.
https://github.com/pytorch/pytorch/blob/7b089c4cd21b046fdae85d0cec62961327bd8cf2/torch/testing/_internal/common_methods_invocations.py#L711-L722

Reply:
I agree with you it's simpler to add these directly for now. Thanks for paying close attention here. This generation is well hidden by method_tests(). If we do end up with similar generation at least it will be explicit in the sample input function.